src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLStopImpl.java

```diff
@@ -244,12 +246,59 @@ public DataFetcher<Iterable<TripTimeOnDate>> stopTimesForPattern() {
             return null;
           }
 
+          TimetableSnapshot timetableSnapshot = transitService.getTimetableSnapshot();
+          long startTime = args.getLegacyGraphQLStartTime();
+          if (timetableSnapshot != null && timetableSnapshot.hasLastAddedTripPatterns()) {
+            LocalDate date = Instant
+              .ofEpochSecond(startTime == 0 ? System.currentTimeMillis() / 1000 : startTime)
+              .atZone(transitService.getTimeZone())
+              .toLocalDate();
+            return Stream
+              .concat(
+                getPatterns(environment)
+                  .stream()
+                  .flatMap(tripPattern ->
+                    tripPattern
+                      .scheduledTripsAsStream()
+                      .map(trip -> timetableSnapshot.getLastAddedTripPattern(trip.getId(), date))
```

Comment:
This way of finding trip patterns created of this requests pattern is a bit inefficient. I don't know if there is a better way to do it without modifying the transit model too much.

Reply:
I think it's okay since it's only one extra lookup, isn't it?

However, could you move the entire block into a method so this one doesn't get so huge?

Reply:
I realized there was an unnecessary loop so now it's a bit better. Also moved some code into a method as suggested.

Reply:
I moved it hopefully properly into a block now in https://github.com/opentripplanner/OpenTripPlanner/pull/4299/commits/0abb304aa3de12690b5e90a2c9c3405f65a9dc42 . I tried looking at if this could be implemented in a prettier way in the StopTimeHelper class but it would have made the logic even a bit more complicated so I didn't end up doing that. I think it's better to rewrite this after/during the realtime part of the transit model refactoring.